### PR TITLE
fix(components-native): Apply input transform in InputText before trim [OPS-20241]

### PR DESCRIPTION
**Pull request by davidr64**
## Motivations

InputNumber was providing string values when text ended with a space.

## Changes

InputText applies the transform before treating the form field value as a string.

## Testing

Use InputNumber finish the number with a space like "13 " and see that InputNumber returns the number 13 instead of the string "13" **after blur**

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

Random photo of Atlantis

### CODE REVIEW

File: packages/components-native/src/InputText/InputText.tsx [hunk 1]
```diff
@@ -444,25 +437,33 @@ function InputTextInternal(
...
+  }
 }
 
 function trimWhitespace(
-  field: ControllerRenderProps<FieldValues, string>,
-  onChangeText?: (newValue: string) => void,
+  inputValue: string | undefined,
+  onChangeText: (newValue: string) => void,
 ) {
-  if (!field.value || !field.value.trim) {
```
Comment (davidr64) [inline packages/components-native/src/InputText/InputText.tsx:453]: Previously field.value could have been a transformed value (most likely a number).

File: packages/components-native/src/InputText/InputText.test.tsx [hunk 2]
```diff
@@ -528,3 +529,104 @@ describe("InputText", () => {
...
+  describe("When working with uncontrolled components", () => {
+    const onFormValueUpdate = jest.fn();
+    beforeEach(jest.clearAllMocks);
+
+    function FormWrapper({ children }: { readonly children: React.ReactNode }) {
+      const form = useForm();
+      useEffect(() => {
+        return form.watch((value, { name, type }) => {
+          console.log(value, name, type);
+          console.log("value", value);
```
Comment (rodrigoeidelvein) [inline packages/components-native/src/InputText/InputText.test.tsx:586]:
```suggestion
```

We can remove these

**Review by rodrigoeidelvein — CHANGES_REQUESTED** [review]
Just a minor cleanup on those console.logs and we're good!

I tested with the pre-release branch, and also applying the changes on the deprecated `atlantis/InputText`, it looks good! Thanks, @davidr64

**Review by rodrigoeidelvein — APPROVED** [review]
LGTM!

File: packages/components-native/src/InputText/InputText.test.tsx [hunk 3]
```diff
@@ -528,3 +529,102 @@ describe("InputText", () => {
...
+  });
+
+  describe("When working with uncontrolled components", () => {
+    const onFormValueUpdate = jest.fn();
+    beforeEach(jest.clearAllMocks);
+
+    function FormWrapper({ children }: { readonly children: React.ReactNode }) {
+      const form = useForm();
+      useEffect(() => {
+        return form.watch((value, { name, type }) => {
```
Comment (rodrigoeidelvein) [inline packages/components-native/src/InputText/InputText.test.tsx:584]:
```suggestion
        return form.watch(value => {
```

**Review by rodrigoeidelvein — CHANGES_REQUESTED** [review]
These params are not being used anymore, that's why it failed

File: packages/components-native/src/InputText/InputText.test.tsx (hunk 3 above)
Comment (davidr64) [reply packages/components-native/src/InputText/InputText.test.tsx:584]: Ah thanks I just pulled to see why

Comment (davidr64) [reply packages/components-native/src/InputText/InputText.test.tsx]: I blame me not paying attention and the linting failure output being obtuse.

**Review by rodrigoeidelvein — APPROVED** [review]
Linting error is gone! :shipit:

**Pull request merged**